Some fixes for tutorials. Also, a Hello World (#663)
stanford-oval/almond-cloud: doc/thingpedia-tutorial-hello-world.md

> # Tutorial 0: Hello World

Reviewer: Should we renumber the tutorials? Or are we hard-core programmers?

Author: Haha sure, I have no preference. I can renumber it

> Welcome to Almond! This tutorial will get you started making your very first Almond device. 

Reviewer: Is the expectation that the user already has a developer account in this tutorial? Is that described elsewhere?

Author: That's on here right? https://almond-dev.stanford.edu/doc/getting-started.md

Author: Added a line to direct users to Getting Started if they don't have a developer account.

## Step 1: set up your device
Go to the [Device Creation Page](/thingpedia/upload/create), fill in the following basic information 
about the device:

> - ID: `hello.<your-name>` (Each device in Thingpedia needs an unique ID. Make sure your ID hasn't been used!)

Reviewer: We should encourage people to follow reverse DNS notation (that is, use a domain they control, such as their github page), with the exception that they can use their own username as the first component.

Author: Okay so `<your-name>.hello`?

- License: `MIT` or any appropriate license
- Icon: choose any PNG file you like (512x512 resolution is recommended)
- JS code: upload a file named `index.js` with the following code.
```javascript
"use strict";

const Tp = require('thingpedia');

module.exports = class MyFirstDevice extends Tp.BaseDevice {
    // A simple query function called "greeting"
    // Query functions must be in the form "get_<something>",
    // where <something> matches the query name in manifest.tt
    get_greeting({ name }) { // `name`` is an optional parameter (see manifest.tt)
        // All queries should return an array of objects
        // The object has properties corresponding to the outputs
        // declared in manifest.tt
        if (name) return [{ reply: name }];
        else return [{ reply: 'world' }];
    }
};
```

## Step 2: describe what your device does
Click on `manifest.tt` on the left panel. 
Copy the following code to the editor and replace `hello.<your-name>` with the 
actual device ID:
```tt
class @hello.<your-name> {
  import loader from @org.thingpedia.v2();
  import config from @org.thingpedia.config.none();

  query greeting(in opt name: String #_[canonical="name"],

>               out reply: String #_[canonical="reply"])

Reviewer: `#_[canonical]` is optional for English

Author: Okay will drop this

  #_[confirmation="greeting"]
  #_[formatted=[{type="text",text="Hello ${reply}!"}]];
}

```

## Step 3: provide some natural language examples
Click on `dataset.tt` on the left panel. 
Copy the following code to the editor and replace `hello.<your-name>` with the 
actual device ID.

> With queries, Almond learns to listen out for commands that goes 'get <something>'. With the `dataset.tt` here, Almond will learn two commands, 'get greeting' and 'get greeting for "<name>"'.

Reviewer: ```suggestion
With queries, Almond learns to listen out for commands that say 'get <something>'. With this `dataset.tt`, Almond will learn two commands: 'get greeting' and 'get greeting for "<name>"'.
```

Also, do we want to highlight the quotation marks around name?

Author: What do you mean highlight?

Author: The quotation marks are important though, this command doesn't work without the quotation marks for now

Reviewer: Yeah I meant adding a sentence that say "note the quotation marks" or similar.

Author: Okay will do
